stdlib/tests/crypto/stark/verifier_recursive/mod.rs

```diff
@@ -28,10 +28,17 @@ pub fn generate_advice_inputs(
     proof: Proof,
     pub_inputs: <ProcessorAir as Air>::PublicInputs,
 ) -> Result<VerifierData, VerifierError> {
+    // we compute the number of procedures in the kernel
+    let pub_inputs_elements = pub_inputs.to_elements();
+    let digests_elements = pub_inputs_elements.len() - MIN_STACK_DEPTH * 2;
+    assert_eq!(digests_elements % WORD_SIZE, 0);
+    let num_digests = digests_elements / WORD_SIZE;
+    let num_kernel_procedures_digests = num_digests - 1;
```

Comment:
Why do we need the `- 1` here? Also why doesn't this underflow when the kernel is empty? I suppose `num_digests` is always at least 1, but I can't see how that would make sense.

Is it worth adding a comment?

Reply:
Should have been more clear, the -1 is to account for the program hash digest.
I have now made this more explicit and added a few comments